src/View/ArrayData.php

```diff
@@ -77,7 +77,7 @@ public function toMap()
     public function getField($field)
     {
         $value = $this->array[$field];
-        if (is_object($value) && !$value instanceof ViewableData) {
+        if (is_object($value) && !($value instanceof ViewableData) && !is_iterable($value)) {
             return new ArrayData($value);
```

Comment:
This is needed because otherwise `ArrayIterator` and other iterators get shoved inside `ArrayData` which is obviously not correct and results in not being able to get the count.